### PR TITLE
Disable FIPS-related rule group for 'osbuild' platform

### LINKED ISSUE #9559: FIPS enabled profiles built in offline mode with image-builder won't boot
#### Description of problem:
Images built with image-builder on-prem using the `ospp`, `cui` & `stig` profiles won't boot. The issue is that these profiles enable fips mode and the images won't boot with `/boot` on a separate partition. A workaround for this is to add `/boot/<PARTITION>` to the kernal arguments.

#### SCAP Security Guide Version:
scap-security-guide-0.1.60-7.el8.noarch

#### Operating System Version:
RHEL 8.7

#### Steps to Reproduce:

1. using [osbuild-composer](https://github.com/osbuild/osbuild-composer) to build an image on-prem, create a blueprint with the following customizations:

```
[customizations.openscap]
profile_id = "ospp"  # or cui
datastream = "/usr/share/xml/scap/ssg/content/ssg-rhel8-ds.xml"
```

2. Attempt to boot the image

#### Actual Results:
The image enters dracut recovery mode

#### Expected Results:
Boot and use the image

#### Additional Information/Debugging Steps:
Remediation logs:
[image-builder-remediation.txt](https://github.com/ComplianceAsCode/content/files/9634398/image-builder-remediation.txt):

Boot log:
[rdsosreport.txt](https://github.com/ComplianceAsCode/content/files/9634399/rdsosreport.txt)

### PR DESCRIPTION
**Pull request by evgenyz**
#### Description:

Do not execute FIPS-related rules in OSBuild environment. Due to the nature of the image creation process the way FIPS should be enabled is different. It is yet to be developed on the IB side but it will be similar to Anakonda's process.

#### Rationale:

- The rules are based on runtime sources of information and therefore can't be used in the OSBuild environment.
- Moreover, the script that is used in remediation also does not support chrooted environment.

- Fixes #9559

### CODE REVIEW

**Comment by jan-cerny** [comment]
The mypy test failure seems unrelated to this PR but I can't find the cause of it, the previous CI runs use the same version of python3-mypy (eg. https://github.com/ComplianceAsCode/content/actions/runs/4233108771/jobs/7353647493) and also the last change in the offending file utils/srg_utils.py was in October.

**Review by jan-cerny — APPROVED** [review]
I have seen in built `ssg-rhel9-ds.xml` that the group `xccdf_org.ssgproject.content_group_fips` has the element `<xccdf-1.2:platform idref="#machine_and_not_osbuild"/>` and the tests look fine

```
[jcerny@thinkpad scap-security-guide{pr/10245}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel9 enable_dracut_fips_module
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-02-22-1259/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_enable_dracut_fips_module
INFO - Script fips_dracut_module_missing.fail.sh using profile (all) OK
INFO - Script fips_dracut_module_present.pass.sh using profile (all) OK
[jcerny@thinkpad scap-security-guide{pr/10245}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel9 --remediate-using ansible enable_dracut_fips_module
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-02-22-1301/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_enable_dracut_fips_module
INFO - Script fips_dracut_module_missing.fail.sh using profile (all) OK
INFO - Script fips_dracut_module_present.pass.sh using profile (all) OK
[jcerny@thinkpad scap-security-guide{pr/10245}]$ 
```

**Comment by jan-cerny** [comment]
@evgenyz We're sorry but we had an issue with the CI jobs and unfortunately they're required to pass before merging. Could you please rebase this PR on the top of the latest upstream master branch? That will include fix from #10250 to this PR and will unblock the CI.

**Review by jan-cerny — APPROVED** [review]

**Comment by hasib-haque** [comment]
Hello, does this issue have a solution or is it still open?

with the boot parameters set to fips=1 as below with openscap STIG, my ec2 instances built from the image drops to dracut prompt. 

blueprint:
```toml
#
# Base definition
#
name = "base-image"
description = "RHEL 8.7 Minimal Install"
version = "2022.12.08"
distro = "rhel-87"

#
# Boot parameters

[customizations.kernel]
append = "no_timer_check console=hvc0 LANG=en_US.UTF-8 console=ttyS0,115200n8 elevator=noop net.ifnames=0  fips=1"

#
# Filesystem
#

[[customizations.filesystem]]
mountpoint = "/boot"
size = 32212254720

[[customizations.filesystem]]
mountpoint = "/var"
size = 4294967296

[[customizations.filesystem]]
mountpoint = "/var/log"
size = 2147483648

# Groups
[[groups]]
name="Minimal Install"

[[packages]]
name = "selinux-policy-targeted"
version = "*"

[[packages]]
name = "crypto-policies"
version = "*"

[[packages]]
name = "crypto-policies-scripts"
version = "*"

[[packages]]
name = "dracut-fips"
version = "*"

#[[packages]]
#name = "dracut-config-generic"
#version = "*"

#[[packages]]
#name = "dracut-network"
#version = "*"

[[packages]]
name = "lvm2"
version = "*"

# Security compliance
#
[[packages]]
name = "aide"
version = "*"
[[packages]]
name = "openscap"
version = "*"
[[packages]]
name = "openscap-scanner"
version = "*"
[[packages]]
name = "scap-security-guide"
version = "*"

[customizations.openscap]
datastream = "/usr/share/xml/scap/ssg/content/ssg-rhel8-ds.xml"
profile_id = "xccdf_org.ssgproject.content_profile_stig"
```

Boot error:
[   19.248546] dracut: Refusing to continue
[   19.254401] dracut Warning: Signal caught!
dracut Warning: Signal caught!
[   19.275879] dracut Warning: dracut: FATAL: FIPS integrity test failed
[   19.275953] dracut Warning: dracut: Refusing to continue
dracut Warning: dracut: FATAL: FIPS integrity test failed
dracut Warning: dracut: Refusing to continue
%G%Gdracut Warning: /boot//.vmlinuz-4.18.0-425.19.2.el8_7.x86_64.hmac does not exist

**Comment by evgenyz** [comment]
Since https://github.com/ComplianceAsCode/content/commit/b34be69dccc62e9fca9277b43c1d08d865f130a9 CaC content should not try to enable the FIPS mode in IB. Until some steps are implemented on the IB side it just does not make sense.

It'd take some time for the updated content to hit RHEL, though.

**Comment by bmoore-ineteng** [comment]
FIPS is a hard requirement for the DISA STIG. Will the image builder tool every be able to produced STIG compliant images?

**Pull request merged**
